public/app/features/alerting/unified/components/contact-points/useNotificationTemplates.ts

```diff
@@ -0,0 +1,138 @@
+import { produce } from 'immer';
+
+import { ActiveTab as ContactPointsActiveTabs } from 'app/features/alerting/unified/components/contact-points/ContactPoints';
+import { useDispatch } from 'app/types';
+
+import { AlertManagerCortexConfig } from '../../../../../plugins/datasource/alertmanager/types';
+import { alertmanagerApi } from '../../api/alertmanagerApi';
+import { updateAlertManagerConfigAction } from '../../state/actions';
+import { ensureDefine } from '../../utils/templates';
+import { TemplateFormValues } from '../receivers/TemplateForm';
+
+// TODO Check the issue with useAlertmanagerConfig hook
+export function useNotificationTemplates(amSourceName?: string) {
+  const templatesRequestState = alertmanagerApi.endpoints.getAlertmanagerConfiguration.useQuery(amSourceName || '', {
```

Comment:
I don't use the `useAlertmanagerConfig` hook here, because it's not possible to use it with the `selectFromResult` option